spacy_llm/pipeline/llm.py

```diff
@@ -178,12 +186,15 @@ def __call__(self, doc: Doc) -> Doc:
         """
         docs = [self._cache[doc]]
         if docs[0] is None:
-            docs = list(
-                self._parse(
-                    [doc],
-                    self._backend(self._template([doc])),
-                )
-            )
+            prompt = list(self._template([doc]))[0]
+            if self._verbose:
+                msg.divider("Generated Prompt")
+                print(prompt)  # noqa T201
+            response = list(self._backend([prompt]))[0]
+            if self._verbose:
+                msg.divider("Response from Backend")
+                print(response)  # noqa T201
+            docs = list(self._parse([doc], [response]))
```

Comment:
- We should `assert` that the converted list has length 1.
- As you mentioned, `print()` isn't great. As `spacy-llm` should be integrated into spaCy soonish, let's use the same logging approach as there?
- Since we have to check for `self._verbose` everytime, it'd be nice to have a helper function `def log(verbose: bool, msg: str)` (maybe in a new `util.py`) that does that so we can log in a single line.

Reply:
* For sure, I'll probably separate this into its own method that operates on batches of docs, then call that method in `__call__` and `pipe` both
* what is the spaCy logging approach? that's not so clear to me. 
* Sure that seems fine to me but want to make sure there's consistency with spaCy

Reply:
> - what is the spaCy logging approach? that's not so clear to me.

IMO using `wasabi` (see the [CLI functionality](https://github.com/explosion/spaCy/blob/master/spacy/cli/evaluate.py#L148)).

> - Sure that seems fine to me but want to make sure there's consistency with spaCy

I think whether we put this in a helper function or not is an implementation detail and doesn't matter that much w.r.t. consistency.

@adrianeboyd's opinion is appreciated here.

Reply:
We'd use `wasabi` for CLI output, but not within pipeline components.

In v3 we tried switching to `logging` loggers in various places to mixed success. We had a lot of trouble getting messages filtered as intended so I remember going back to python warnings in a number of places. We still have logging for training, but basically nowhere else in the library. (I see a few traces that I think aren't intentional.)

To me this feels like it should be using logging and be behind a flag, so if it's noisy then users can easily disable all of it at once. (Maybe there are some obvious solutions that we missed to configuring logging before, but I just remember none of it working like we'd intended for warnings in particular.)

And I would add that I'm also not sure how I feel about adding `verbose` as a component-level config option.

Reply:
So maybe just properly configured logging is what we actually want? I've just not had any success in the library context with this...

Reply:
> I've just not had any success in the library context with this...

How so?

Reply:
The main problems were related to warnings. We couldn't get warnings working the way we wanted and the combination of having both python warnings and logger warnings (where users couldn't really tell the difference) was not good.

If this is below `WARNING` maybe it's not as much of an issue? I'm not sure how configurable you would need this to be in the end. For the training loggers it makes sense to have a bit more structure, but I'm not sure inference loggers need much more than what's provided by `logging`?